### DISCUSSION TITLE
VanJS with Web Components & Shadow DOM?

### DISCUSSION THREAD

**Discussion by davidhmays** [vanjs-org/van#276] [Q&A] [answered]
Sorry for the dumb question, but could VanJS components also be web components with a Shadow DOM for full CSS encapsulation?
It doesn't seem like there would be a reason not, but having tried this with another framework and run into issues being unable to add child components, thought I'd ask first.

**Comment by sirenkovladd** [comment]
You can try [van-element](https://github.com/Atmos4/van-element)

> isolated styles

**Reply by sirenkovladd** [reply, marked as answer]
Here is an example
https://van-element.pages.dev/learn/styles.html#style-tag

**Reply by davidhmays** [reply]
Oh awesome, thanks!
